lib/msf/sanity.rb

```diff
@@ -3,36 +3,12 @@
 # Provides some sanity checks against the ruby build and version
 #
 
-# Check for the broken pack/unpack in OS X 10.4.x
-if ([1].pack('n') == "\x01\x00")
-  $stderr.puts "*** This ruby build has a broken pack/unpack implementation! "
-
-  if (RUBY_PLATFORM =~ /darwin/)
-    $stderr.puts "    Apple shipped a broken version of ruby with the 10.4.x   "
-    $stderr.puts "    release. Please install ruby from source, or use one of  "
-    $stderr.puts "    the free package managers to obtain a working ruby build."
-  end
-
-  exit(0)
-end
-
-# Check for ruby 1.8.2 as the minimal supported version
-if (RUBY_VERSION =~ /^1\.[0-7]\./ or RUBY_VERSION =~ /^1\.8\.[0-1]$/)
-  $stderr.puts "*** This version of ruby is not supported, please upgrade to 1.8.7+"
+# Check for ruby 1.x and throw a warning
+if (RUBY_VERSION =~ /^1/)
+  $stderr.puts "*** Ruby 1.x is not supported, please upgrade to Ruby 2.1 or newer."
   exit(0)
```

Comment:
This should probably non-zero. Then it can be folded into the above line as: `abort '*** Ruby 1.x is not supported, please upgrade to Ruby 2.1 or newer'`.